dandiapi/api/tasks.py

```diff
@@ -164,3 +165,57 @@ def validate_version_metadata(version_id: int) -> None:
 @shared_task
 def delete_doi_task(doi: str) -> None:
     delete_doi(doi)
+
+
+@shared_task
+def unembargo_dandiset(dandiset_id: int):
+    """Unembargo a dandiset by copying all embargoed asset blobs to the public bucket."""
+    dandiset: Dandiset = Dandiset.objects.get(id=dandiset_id)
+
+    # Only the draft version is needed, since embargoed dandisets can't be published
+    draft_version: Version = dandiset.draft_version
+    embargoed_assets: List[Asset] = list(
+        Asset.objects.filter(versions__in=[draft_version], embargoed_blob__isnull=False)
+    )
+
+    # Copy all embargoed assets to public bucket
+    for asset in embargoed_assets:
+        etag = asset.embargoed_blob.etag
+        matching_blob = AssetBlob.objects.filter(etag=etag).first()
+
+        # Use existing AssetBlob if possible
+        if matching_blob is not None:
+            asset.blob = matching_blob
+        else:
+            # Matching AssetBlob doesn't exist, copy blob to public bucket
+            resp = copy_object(
+                asset.embargoed_blob.blob.storage,
+                source_bucket=settings.DANDI_DANDISETS_EMBARGO_BUCKET_NAME,
+                source_key=asset.embargoed_blob.blob.name,
+                dest_bucket=settings.DANDI_DANDISETS_BUCKET_NAME,
+                dest_key=Upload.object_key(asset.embargoed_blob.blob_id, dandiset),
+            )
+
+            # Assert files are equal
+            assert resp.etag == asset.embargoed_blob.etag
+
+            # Assign blob
+            asset.blob = AssetBlob(
+                blob_id=asset.embargoed_blob.blob_id,
+                etag=resp.etag,
+                blob=resp.key,
+                size=asset.embargoed_blob.size,
+            )
+            asset.blob.save()
```

Comment:
You might consider moving this logic to a method on `EmbargoedAssetBlob` to keep as much application logic on the model as possible, but not a big deal if you think here is better